libdevcore/CommonData.cpp

```diff
@@ -103,3 +103,26 @@ bool dev::passesAddressChecksum(string const& _str, bool _strict)
 	}
 	return true;
 }
+
+std::string dev::getChecksumAddress(std::string const& _addr)
```

Comment:
This could use some refactor and common parts be shared between this and `passesAddressChecksum`.

Reply:
Sure, I'll push something up in a moment. One thing I think I could do as well, is just get rid of the check to see if addressCharacter is a number or not - according to isupper docs, it'll return the same character if a number (or something without an uppercase component) is passed in. I wasn't sure if I should leave the check in for clarity though.